### ISSUE TITLE
visit fails when running in 'run', but succeeds when running in 'open'

### ISSUE THREAD

**Issue by yonatanaftali** [cypress-io/cypress#18105]
### Current behavior

I have a test that validates some value in the IndexedDB. I'm using [idb-keyval](https://www.npmjs.com/package/idb-keyval) to interact with the IDB.

In that test, I have a function that receives an expected value and compares it with the value in the IndexedDB. Please see the attached code below.

The test works fine if I run Cypress in 'open' mode. However, if I run Cypress in 'run' mode, the test fails on the visit command.

### Desired behavior

Cypress open and run should behave the same and execute the visit command.

### Test code to reproduce

```js
import { get, createStore } from 'idb-keyval';
import { DB_NAME } from '@/indexedDb';

const customStore = createStore(DB_NAME, 'state'); // required for idb-keyval

describe('Avatar Menu', () => {
  before(() => {
    cy.visit('/');
  });

  it('Check active extension', () => {
    get('localSettings', customStore).then(localSettings => {
      const activeExtension = localSettings["activeExtension"];
      expect(activeExtension).to.equal('200');
    })
  });
});
```

### Cypress Version

8.4.1

**Comment by jennifer-shehane** [comment]
- What is the error message that's shown during `cypress run`?
- `cypress run` runs in Electron by default, have you tried running in Electron during `cypress open`? Do you see the same failure?

**Comment by yonatanaftali** [comment]
@jennifer-shehane 

- I'm getting the following error: `TypeError: The following error originated from your test code, not from Cypress. It was caused by an unhandled promise rejection. > Cannot read property 'activeExtension' of undefined`. Also, I'm seeing a white screen, since the visit was unsuccessful.

- I did, and it works fine. The only scenario in which I the visit fails for me is when doing `cypress run`.
